packages/nextra-theme-docs/src/search.tsx

```diff
@@ -77,15 +77,24 @@ const Search = ({ directories = [] }: SearchProps) => {
         moveActiveItem(UP)
       }
 
-      if (key === 'Enter' && results && results[active]) {
+      if (active !== null && key === 'Enter' && results && results[active]) {
         router.push(results[active].route)
       }
     },
     [active, results, router]
   )
 
+  const handleOnBlur = useCallback(
+    e => {
+      if (active === null) {
+        setShow(false)
+      }
+    },
+    [active]
```

Comment:
Note: this is the core issue here ^: currently, when a user selects a search result option using their mouse, the menu option immediately gets un-rendered and the actual user click does not get triggered to actually capture the re-route request. I took this approach to minimize any serious refactor in the current implementation. I did however, apply some basic HTML structure cleanup in the process within this file.